main.cc

```diff
@@ -1259,6 +1252,13 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             qp.start(std::ref(proxy), std::ref(forward_service), std::move(local_data_dict), std::ref(mm_notifier), std::ref(mm), qp_mcfg, std::ref(cql_config), std::move(auth_prep_cache_config), std::ref(group0_client), std::move(wasm_ctx)).get();
 
+            supervisor::notify("creating tracing");
+            sharded<tracing::tracing>& tracing = tracing::tracing::tracing_instance();
+            tracing.start(sstring("trace_keyspace_helper")).get();
+            auto destroy_tracing = defer_verbose_shutdown("tracing instance", [&tracing] {
+                tracing.stop().get();
+            });
+
```

Comment:
Note that traced sessions can remain as long as the replica rpc server (storage_proxy server side) is alive. A tracing session can start on some remote coordinator and propagate via rpc.

tracing can only be stopped after storage_proxy (server side) is drained. In turn, tracing depends on storage_proxy (client side) to drain, though we can drain tracing by cancelling everything.

So I think some more thought it needed here.

Reply:
By the time this defer happens both `proxy.drain_on_shutdown()` and `proxy.uninit_messaging_service` are called, so proxy is isolated from the network and (expected to be) drained. And _after_ this defer there's no meaningful `proxy.stop()` ("no meaningful" means -- there's commented-out `sharded<storage_proxy>.stop()` and `storage_proxy::stop()` is no-op) . So what's there in proxy that should be stopped before tracing stops here?

Reply:
Ah, also note, that tracing is _drained_ much earlier -- in storage_service::do_drain() which's called first via drain_on_shutdown defer.